todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -33,43 +37,41 @@ class TasksLocalDataSource private constructor(
      * Note: [TasksDataSource.LoadTasksCallback.onDataNotAvailable] is fired if the database doesn't exist
      * or the table is empty.
      */
-    override fun getTasks(callback: TasksDataSource.LoadTasksCallback) {
-        appExecutors.diskIO.execute {
+    override suspend fun getTasks(): List<Task> {
```

Comment:
`async/await` is designed to be used when you want to return `Deferred`. In this case it's better to use `withContext`. 

Also, do not switch to UI thread, it's up to client when to switch context. See https://github.com/dmytrodanylyk/android-architecture/blob/dev-todo-mvp-kotlin-coroutines/todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt#L36